clientv3/client.go

```diff
@@ -169,6 +169,7 @@ func (c *Client) retryConnection(oldConn *grpc.ClientConn, err error) (*grpc.Cli
 		// conn has already been updated
 		return c.conn, nil
 	}
+	oldConn.Close()
```

Comment:
`oldConn` should already be closed/lost before calling retryConnection(), otherwise use of this function could devolve into rapidly flipping the connection on and off